python/pyarrow/types.pxi

```diff
@@ -3173,6 +3215,9 @@ def unify_schemas(schemas):
     ----------
     schemas : list of Schema
         Schemas to merge into a single one.
+    options : FieldMergeOptions or string, optional
```

Comment:
Maybe we should align the keyword name with the other places where we pass this? In `concat_tables`, it's called `field_merge_options`, which is quite long, so another option is something in between (`merge_options`, although that's probably not very descriptive being about the field types, or `promote_options`, but then that deviates from the naming in C++)

Reply:
I can rename it to `promote_options` everywhere.

Reply:
Hmm, maybe this 🚢 has sailed already. It is also called options in `cast(target_schema, safe, options)`, I like the consistency there

Reply:
Yes, but for the cast example, it are the CastOptions passed to the cast kernel. While here we have options for promoting types that is not tied to a specific function (i.e. it's used both in unify_schema, concat_tables, and potentially others)